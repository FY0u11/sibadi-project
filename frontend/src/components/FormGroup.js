import React from 'react'

const FormGroup = ({ type, name, label, message, value, setValue }) => {
  return (
    <div className="form-group">
      <label htmlFor={name}>
        {label}
        <strong> {message}</strong>
      </label>
      <input
        id={name}
        type={type}
        className="form-control"
        placeholder={`Enter ${label.toLowerCase()}`}
        value={value}
        onChange={event => setValue(`${name}Field`, event.target.value)}
      />
    </div>
  )
}

export default FormGroup
