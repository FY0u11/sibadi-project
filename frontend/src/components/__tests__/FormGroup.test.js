import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import FormGroup from '../FormGroup'

afterEach(cleanup)

const props = {
  type: 'text',
  name: 'test',
  label: 'test label',
  message: 'test message',
  value: 'test value'
}

describe('Testing FormGroup component', () => {
  it('It should be properly rendered', async () => {
    const { asFragment } = render(<FormGroup {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('It should trigger callback with new input value on change event', () => {
    const callback = jest.fn()
    const { getByDisplayValue } = render(<FormGroup {...props} setValue={callback} />)
    fireEvent.change(getByDisplayValue('test value'), { target: { value: 'value 42' } })
    expect(callback).toBeCalledTimes(1)
    expect(callback).toBeCalledWith('testField', 'value 42')
  })
})
