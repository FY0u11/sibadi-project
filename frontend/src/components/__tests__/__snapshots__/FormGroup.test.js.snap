// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FormGroup component It should be properly rendered 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      for="test"
    >
      test label
      <strong>
         test message
      </strong>
    </label>
    <input
      class="form-control"
      id="test"
      placeholder="Enter test label"
      type="text"
      value="test value"
    />
  </div>
</DocumentFragment>
`;
